extensions/wikia/AdEngine/js/provider/gpt/adSizeMap.js

```diff
@@ -0,0 +1,57 @@
+/*global define*/
+define('ext.wikia.adEngine.provider.gpt.adSizeMap', [
+	'wikia.window',
+	'wikia.log'
+], function (win, log) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.provider.gpt.adSizeMap';
+
+	function SizeMap(slotSizes) {
+		log(['SizeMap', slotSizes], 'debug', logGroup);
+		this.mapping = slotSizes || [];
+	}
+
+	SizeMap.is = function (object) {
+		return object instanceof SizeMap;
+	};
+
+	SizeMap.prototype.toGptSizeMap = function () {
+		log(['toGptSizeMap', this.mapping], 'debug', logGroup);
+		var sizeMapping = win.googletag && win.googletag.sizeMapping();
+
+		if (!sizeMapping) {
```

Comment:
Is it safe from timing perspective? `gpt.helper::pushAd` method isn't wrapped with googletag.cmd so it may not work for TOP_LEADERBOARD, because we load gpt library with the first push

https://github.com/Wikia/app/blob/dev/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L166